logstash-core-event-java/src/main/java/com/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -235,26 +229,21 @@ public IRubyObject ruby_to_s(ThreadContext context)
         @JRubyMethod(name = "to_hash")
         public IRubyObject ruby_to_hash(ThreadContext context) throws IOException
         {
-            // TODO: is this the most efficient?
-            RubyHash hash = JavaUtil.convertJavaToUsableRubyObject(context.runtime, this.event.toMap()).convertToHash();
-            // inject RubyTimestamp in new hash
-            hash.put(PathCache.TIMESTAMP, JrubyTimestampExtLibrary.RubyTimestamp.newRubyTimestamp(context.runtime, this.event.getTimestamp()));
-            return hash;
+            return Rubyfier.deep(context.runtime, this.event.toMap());
         }
 
         @JRubyMethod(name = "to_hash_with_metadata")
         public IRubyObject ruby_to_hash_with_metadata(ThreadContext context) throws IOException
         {
-            HashMap<String, Object> dataAndMetadata = new HashMap<String, Object>(this.event.getData());
+            Map data;
+
             if (!this.event.getMetadata().isEmpty()) {
-                dataAndMetadata.put(Event.METADATA, this.event.getMetadata());
+                data = new HashMap<>(this.event.toMap());
+                data.put(Event.METADATA, this.event.getMetadata());
+            } else {
+                data = this.event.toMap();
```

Comment:
I might be mistaken, but could this else clause be eliminated by, in line 238, setting data to `this.event.toMap()` and removing line 241?